sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -113,221 +115,306 @@ void DistortionCorrectorComponent::onImu(const sensor_msgs::msg::Imu::ConstShare
+    imu_stamp = rclcpp::Time(it_imu->header.stamp).seconds();
   }
 
-  // If there is a point that cannot be associated, record it to issue a warning
-  bool twist_time_stamp_is_too_late = false;
-  bool imu_time_stamp_is_too_late = false;
+  // If there is a point in a pointcloud that cannot be associated, record it to issue a warning
+  bool is_twist_time_stamp_too_late = false;
+  bool is_imu_time_stamp_is_too_late = false;
+  bool is_twist_valid = true;
+  bool is_imu_valid = true;
 
   for (; it_x != it_x.end(); ++it_x, ++it_y, ++it_z, ++it_time_stamp) {
-    while (twist_it != std::end(twist_queue_) - 1 && *it_time_stamp > twist_stamp) {
-      ++twist_it;
-      twist_stamp = rclcpp::Time(twist_it->header.stamp).seconds();
-    }
-
-    float v{static_cast<float>(twist_it->twist.linear.x)};
-    float w{static_cast<float>(twist_it->twist.angular.z)};
+    is_twist_valid = true;
+    is_imu_valid = true;
 
+    // Get closest twist information
+    while (it_twist != std::end(twist_queue_) - 1 && *it_time_stamp > twist_stamp) {
+      ++it_twist;
+      twist_stamp = rclcpp::Time(it_twist->header.stamp).seconds();
+    }
     if (std::abs(*it_time_stamp - twist_stamp) > 0.1) {
-      twist_time_stamp_is_too_late = true;
-      v = 0.0f;
-      w = 0.0f;
+      is_twist_time_stamp_too_late = true;
+      is_twist_valid = false;
     }
 
-    if (use_imu_ && !angular_velocity_queue_.empty()) {
-      while (imu_it != std::end(angular_velocity_queue_) - 1 && *it_time_stamp > imu_stamp) {
-        ++imu_it;
-        imu_stamp = rclcpp::Time(imu_it->header.stamp).seconds();
+    // Get closest IMU information
+    if (use_imu && !angular_velocity_queue_.empty()) {
+      while (it_imu != std::end(angular_velocity_queue_) - 1 && *it_time_stamp > imu_stamp) {
+        ++it_imu;
+        imu_stamp = rclcpp::Time(it_imu->header.stamp).seconds();
       }
 
       if (std::abs(*it_time_stamp - imu_stamp) > 0.1) {
-        imu_time_stamp_is_too_late = true;
-      } else {
-        w = static_cast<float>(imu_it->vector.z);
+        is_imu_time_stamp_is_too_late = true;
+        is_imu_valid = false;
       }
+    } else {
+      is_imu_valid = false;
     }
 
-    const auto time_offset = static_cast<float>(*it_time_stamp - prev_time_stamp_sec);
+    float time_offset = static_cast<float>(*it_time_stamp - prev_time_stamp_sec);
 
-    point.setValue(*it_x, *it_y, *it_z);
+    // Undistorted a single point based on the strategy
+    undistortPoint(it_x, it_y, it_z, it_twist, it_imu, time_offset, is_twist_valid, is_imu_valid);
 
-    if (need_transform) {
-      point = tf2_base_link_to_sensor_inv * point;
-    }
+    prev_time_stamp_sec = *it_time_stamp;
+  }
+
+  warnIfTimestampsTooLate(is_twist_time_stamp_too_late, is_imu_time_stamp_is_too_late);
+}
+
+template <class Derived>
+void DistortionCorrector<Derived>::warnIfTimestampsTooLate(
+  bool is_twist_time_stamp_too_late, bool is_imu_time_stamp_is_too_late)
```

Comment:
These names too - the first one is OK, but `is_imu_time_is_too_late` is confusing and also gramatically wrong. I would simplify these also.

Reply:
Thanks! 
I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/7137/commits/f3ab5e80cb80a95f05c910582c05da2972d77b0f#diff-6574a5f6144786166bb66db08ec3da2752dd30581cb573b825e87a9b2b4238af. 

Which also includes changes to your previous suggestions.